src/main/java/com/icthh/xm/tmf/ms/communication/service/firebase/FirebaseService.java

```diff
@@ -146,30 +130,38 @@ private FirebaseMessaging getFirebaseMessaging(CommunicationMessage message) {
             .orElseThrow(() -> new BusinessException("error.fcm.sender.id.invalid", "Sender id is not valid"));
     }
 
-    private CommunicationMessage buildCommunicationResponse(BatchResponse batchResponse, CommunicationMessage message) {
+    private CommunicationMessage buildCommunicationResponse(BatchResponse batchResponse, CommunicationMessage msg,
+                                                            List<Receiver> receivers) {
+        ExtendedCommunicationMessage message = ExtendedCommunicationMessageFactory.newMessage(msg);
 
+        List<Detail> details = new ArrayList<>();
         List<SendResponse> responses = batchResponse.getResponses();
-        List<CommunicationRequestCharacteristic> characteristics = new ArrayList<>(responses.size());
-        characteristics.add(new CommunicationRequestCharacteristic().name("successCount").value(String.valueOf(batchResponse.getSuccessCount())));
-        characteristics.add(new CommunicationRequestCharacteristic().name("failureCount").value(String.valueOf(batchResponse.getFailureCount())));
+        for (int i = 0; i < responses.size(); i++) {
+            SendResponse response = responses.get(i);
+
+            Detail detail = new Detail()
+                .receiver(receivers.get(i));
```

Comment:
Strange question, but the number of responses is always equal to the number of recipients..?

Reply:
Correct. That's what written in the doc:

> The responses list obtained by calling {@link BatchResponse#getResponses()} on the return value corresponds to the order of tokens in the {@link MulticastMessage}.